### PR TITLE
Simplifiers and formatting

### PR DESCRIPTION
**Pull request by d-xo**
## Description

This adds a few simplifers & formatting rules that I wrote while debugging `prove_transfer` today. It needs https://github.com/ethereum/hevm/pull/90 to be merged first.

## Checklist

- [x] tested locally
- [ ] added automated tests
- [ ] updated the docs
- [ ] updated the changelog

### CODE REVIEW

**Comment by d-xo** [comment]
Turns out I should learn to run the test suite before I submit a PR....

File: src/hevm/src/EVM/Expr.hs [hunk 1]
```diff
@@ -487,21 +511,28 @@ stripWrites bottom top = \case
...
+  _ -> Just $ SLoad addr slot store
+readStorage addr' slot' s@AbstractStore = Just $ SLoad addr' slot' s
+readStorage addr' slot' s@(SStore addr slot val prev) =
+  if addr == addr'
+  then if slot == slot'
+       -- if address and slot match then we return the val in this write
+       then Just val
+       else case (slot, slot') of
+              -- if the slots don't match and are lits, we can skip this write
+              (Lit _, Lit _) -> readStorage addr' slot' prev
```
[the remark this answers is not in the document]
Comment (d-xo) [reply src/hevm/src/EVM/Expr.hs:530]: In this case we know the addresses match syntactically so I think it's enough to just check that the slots are Lit to skip the write?

[the remark this answers is not in the document]
Comment (zoep) [reply src/hevm/src/EVM/Expr.hs:530]: oh yes, of course!

**Review by msooseth — APPROVED** [review]
I checked, and it looks okay to me. However, I was also taking part when writing it so my view may be biased.

**Pull request merged**